router/core/ratelimiter.go

```diff
@@ -60,11 +82,35 @@ func (c *CosmoRateLimiter) RateLimitPreFetch(ctx *resolve.Context, info *resolve
 	return &resolve.RateLimitDeny{}, nil
 }
 
+func (c *CosmoRateLimiter) generateKey(ctx *resolve.Context) string {
+	if c.keySuffixFromRequestHeader {
+		v := ctx.Request.Header.Get(c.requestHeaderName)
+		if v != "" {
+			return fmt.Sprintf("%s:%s", ctx.RateLimitOptions.RateLimitKey, v)
```

Comment:
We should validate the value. Enforcing a max length and strip leading whitespaces from it.